src/io/s3_filesys.cc

```diff
@@ -824,10 +824,10 @@ void ListObjects(const URI &path,
 S3FileSystem::S3FileSystem() {
   const char *keyid = getenv("AWS_ACCESS_KEY_ID");
   const char *seckey = getenv("AWS_SECRET_ACCESS_KEY");
-  if (keyid == NULL) {
+  if (keyid == NULL || !*keyid) {
     LOG(FATAL) << "Need to set enviroment variable AWS_ACCESS_KEY_ID to use S3";
   }
-  if (seckey == NULL) {
+  if (seckey == NULL || !*seckey) {
```

Comment:
By googling that, I see cases of some implementations returning an empty string.

Also, this example does not appear to check:  
http://cvsweb.netbsd.org/bsdweb.cgi/src/lib/libc/stdlib/getenv.c?rev=1.19&content-type=text/x-cvsweb-markup